tower-http/src/compression/predicate.rs

```diff
@@ -0,0 +1,242 @@
+//! Predicates for disabling compression of responses.
+//!
+//! Predicates are applied with [`Compression::compress_when`] or
+//! [`CompressionLayer::compress_when`].
+//!
+//! [`Compression::compress_when`]: super::Compression::compress_when
+//! [`CompressionLayer::compress_when`]: super::CompressionLayer::compress_when
+
+use http::{header, Extensions, HeaderMap, StatusCode, Version};
+use http_body::Body;
+use std::{fmt, sync::Arc};
+
+/// Predicate used to determine if a response should be compressed or not.
+pub trait Predicate: Clone {
+    /// Should this response be compressed or not?
+    fn should_compress<B>(&self, response: &http::Response<B>) -> bool
```

Comment:
Note to self: This should probably take `&mut self` as per [`tower::filter::Predicate`](https://docs.rs/tower/0.4.11/tower/filter/trait.Predicate.html)